### PR TITLE
[DO NOT REVIEW] Fixes #9310 - Fixing the containers API

### PR DESCRIPTION
**Pull request by daviddavis**
- [ ] Fix todo items
- [ ] Handle Fog errors
- [ ] Write tests
- [ ] Check api doc

### CODE REVIEW

File: app/controllers/api/v2/containers_controller.rb [hunk 1]
```diff
@@ -18,10 +18,11 @@ class ContainersController < ::Api::V2::BaseController
 
       def index
         if params[:compute_resource_id].present?
-          @containers = Container.where(:compute_resource_id => params[:compute_resource_id])
+          scope = Container.where(:compute_resource_id => params[:compute_resource_id])
         else
-          @containers = Container.all
+          scope = Container.scoped
         end
+        @containers = scope.search_for(params[:search], :order => params[:order]).paginate(:page => params[:page])
```
Comment (daviddavis) [inline app/controllers/api/v2/containers_controller.rb]: Need to document `search`.

File: app/controllers/api/v2/containers_controller.rb [hunk 2]
```diff
@@ -44,7 +45,7 @@ def show
                                             Format should be repository:tag, e.g: centos:7')
           param :tty, :bool
           param :entrypoint, String
-          param :cmd, String
+          param :command, String, :required => true
```
Comment (daviddavis) [inline app/controllers/api/v2/containers_controller.rb]: Docker seems to need a command. Also, the wizard code calls it `command` and not `cmd`.

File: app/controllers/api/v2/containers_controller.rb [hunk 3]
```diff
@@ -125,18 +125,18 @@ def power
 
       def set_wizard_state
         wizard_properties = { :preliminary   => [:compute_resource_id],
-                              :image         => [:registry_id, :repository_name, :tag, :katello],
+                              :image         => [:registry_id, :repository_name, :tag, :image],
```
Comment (daviddavis) [inline app/controllers/api/v2/containers_controller.rb]: Should we remove `repository_name` and `tag`? I don't see them in the apidoc.

File: app/models/docker_container_wizard_states/image.rb [hunk 4]
```diff
@@ -7,5 +7,14 @@ class Image < ActiveRecord::Base
 
     validates :tag,             :presence => true
     validates :repository_name, :presence => true
+
+    def image=(image)
```
Comment (daviddavis) [inline app/models/docker_container_wizard_states/image.rb]: Need tests for this method.

**Comment by daviddavis** [comment]
Also, need to handle when image is not found.

**Comment by eLobato** [comment]
@daviddavis closed by mistake?

**Comment by daviddavis** [comment]
@eLobato no. some of these fixes are in https://github.com/theforeman/foreman-docker/pull/93 and the search param is in https://github.com/theforeman/foreman-docker/pull/94.

**Pull request closed without merging**